cas-server-support-osf/src/main/java/io/cos/cas/authentication/handler/support/OpenScienceFrameworkPrincipalFromRequestRemoteUserNonInteractiveCredentialsAction.java

```diff
@@ -245,6 +245,13 @@ protected Event doExecute(final RequestContext context) throws Exception {
             onSuccess(context, credential);
             return success();
         } catch (final AuthenticationException e) {
+
+            // This is a temporary hotfix to prevent invalid verification from looping indefinitely in the login flow
+            final HttpServletRequest request = WebUtils.getHttpServletRequest(context);
+            if (request.getParameter("verification_key") != null || credential.getVerificationKey() != null) {
+                return new Event(this, "invalidVerificationKey");
```

Comment:
When the verification key is present, instead of throwing an exception, just return the event "invalidVerificationKey".